handlers.js

```diff
@@ -263,11 +263,20 @@ module.exports = function(port) {
 
         var identityCheckParams;
         if (request.payload.method === 'identity.check') {
-            identityCheckParams = assign({}, request.payload.params, {ip: (port.config.allowXFF && request.headers['x-forwarded-for'])
-            ? request.headers['x-forwarded-for'] : request.info.remoteAddress}, request.auth.credentials);
+            identityCheckParams = assign({}, request.payload.params);
         } else {
-            identityCheckParams = assign({actionId: request.payload.method, ip: request.info.remoteAddress}, request.auth.credentials);
+            identityCheckParams = {actionId: request.payload.method};
         }
+        identityCheckParams = assign(
```

Comment:
Your example is not the same.
Look at the first line you have x= , i.e. you are changing the original variable and doing unnecessary extra step through empty object.